include/sound/sof.h

```diff
@@ -39,6 +39,16 @@ enum sof_fw_state {
 	SOF_FW_CRASHED,
 };
 
+/* DSP power states */
+enum sof_dsp_power_states {
+	SOF_DSP_PM_D0,
+	SOF_DSP_PM_D1,
+	SOF_DSP_PM_D2,
+	SOF_DSP_PM_D3_HOT,
+	SOF_DSP_PM_D3,
+	SOF_DSP_PM_D3_COLD,
+};
+
```

Comment:
that move isn't clear to me, why would it be required to know the power state outside of the parent?

Reply:
while I am at it, why do we have D3_HOT, D3 and D3_COLD? It's an existing definition but I don't get why we needed 3 enums for D3.

Reply:
Only the core knows what is the depth that we are going to suspend for example, but a client might need to do different actions depending on the state. S3 vs s2idle for example when the DSP is off or it remains on.
It is simpler to leave it to the clients than invent some obscure flagging to tell the clients what to do.

Reply:
> while I am at it, why do we have D3_HOT, D3 and D3_COLD? It's an existing definition but I don't get why we needed 3 enums for D3.

I just moved these existing definitions, I don't really know the background of these

Reply:
> Only the core knows what is the depth that we are going to suspend for example, but a client might need to do different actions depending on the state. S3 vs s2idle for example when the DSP is off or it remains on. It is simpler to leave it to the clients than invent some obscure flagging to tell the clients what to do.

Yes but this doesn't really work: you would also need to pass the D0 substate to know if the DSP is active or not, but in a non-reachable way (D0ix).